packages/evm/contracts/registry/GuardiansRegistry.sol

```diff
@@ -168,78 +169,82 @@ contract GuardiansRegistry is ControlledRecoverable, IGuardiansRegistry, IERC900
-    * @param _amount Amount of guardian tokens to be deactivated for the next term
+    * @notice Unstake `@tokenAmount(self.token(), _amount)` from `_guardian`
+    * @param _guardian Address of the guardian to unstake tokens from
+    * @param _amount Amount of tokens to be unstaked
+    * @param _data Optional data is never used by this function, only logged
     */
-    function deactivate(uint256 _amount) external {
-        _deactivateTokens(msg.sender, _amount);
+    function unstake(address _guardian, uint256 _amount, bytes calldata _data) external authenticate(_guardian) {
+        _unstake(_guardian, _amount, _data);
     }
 
     /**
-    * @notice Stake `@tokenAmount(self.token(), _amount)` for the sender to the Protocol
-    * @param _amount Amount of tokens to be staked
-    * @param _data Optional data that can be used to request the activation of the transferred tokens
+    * @notice Activate `@tokenAmount(self.token(), _amount)` for `_guardian`
+    * @param _guardian Address of the guardian activating the tokens for
+    * @param _amount Amount of guardian tokens to be activated for the next term
     */
-    function stake(uint256 _amount, bytes calldata _data) external {
-        _stake(msg.sender, msg.sender, _amount, _data);
+    function activate(address _guardian, uint256 _amount) external authenticate(_guardian) {
+        _activate(_guardian, _amount);
     }
 
     /**
-    * @notice Stake `@tokenAmount(self.token(), _amount)` for `_to` to the Protocol
-    * @param _to Address to stake an amount of tokens to
-    * @param _amount Amount of tokens to be staked
-    * @param _data Optional data that can be used to request the activation of the transferred tokens
+    * @notice Deactivate `_amount == 0 ? 'all unlocked tokens' : @tokenAmount(self.token(), _amount)` for `_guardian`
+    * @param _guardian Address of the guardian deactivating the tokens for
+    * @param _amount Amount of guardian tokens to be deactivated for the next term
     */
-    function stakeFor(address _to, uint256 _amount, bytes calldata _data) external {
-        _stake(msg.sender, _to, _amount, _data);
+    function deactivate(address _guardian, uint256 _amount) external authenticate(_guardian) {
+        _deactivate(_guardian, _amount);
     }
 
     /**
-    * @notice Unstake `@tokenAmount(self.token(), _amount)` for `_to` from the Protocol
-    * @param _amount Amount of tokens to be unstaked
+    * @notice Stake and activate `@tokenAmount(self.token(), _amount)` for `_guardian`
+    * @param _guardian Address of the guardian staking and activating tokens for
+    * @param _amount Amount of tokens to be staked and activated
     * @param _data Optional data is never used by this function, only logged
     */
-    function unstake(uint256 _amount, bytes calldata _data) external {
-        _unstake(msg.sender, _amount, _data);
-    }
+    function stakeAndActivate(address _guardian, uint256 _amount, bytes calldata _data) external {
```

Comment:
I'm having a hard time understanding the `stakeAndActivate()` flow, and how we expect to support lock managers with it. Let's discuss!